modules/serverless/pages/serving/scaleability-and-performance-of-serving.adoc

```diff
@@ -0,0 +1,271 @@
+
+
+== Scalability and performance of {serverlessproductname} Serving
+
+{serverlessproductname} Serving has to be scaled and configured based on the following parameters:
+
+* Number of Knative Services
+* Number of Revisions
+* Amount of concurrent requests in the system
+* Size of payloads of the requests
+* The startup-latency and response latency of the Knative Service added by the user's web application
+* Number of changes of the KnativeService `CustomResources` over time
+
+Scaling of {serverlessproductname} Serving is configured using the `KnativeServing` `CustomResource`.
+
+
+=== `KnativeServing` defaults
+
+Per default, {serverlessproductname} Serving is configured to run all components with high availability and with medium-sized CPU and memory requests/limits.
+This also means, that the `high-available` field in `KnativeServing` is automatically set to a value of 2 and all system components are scaled to two relicas.
+This set-up is suitable for medium-sized workload scenarios. These defaults have been tested with
+
+* 170 Knative Services
+* 1-2 Revisions per Knative Service
+* 89 test scenarios mainly focussed on testing the control-plane
+* 48 re-creating scenarios where Knative Services are deleted and re-created
+* 41 stable scenarios, in which requests are slowly but continuously sent to the system
+
+During these test cases, the system components effectively consumed:
+
+|===
+| Component | Measured resources
+
+| Operator in project `openshift-serverless`
+| 1 GB Memory, 0.2 Cores of CPU
+
+| Serving components in project `knative-serving`
+| 5 GB Memory, 2.5 Cores of CPU
+|===
+
+While the default set-up is suitable for medium-sized workloads, this might be over-sized for smaller set-ups or under-sized for high-workload scenarios.
+Please see the next sections for possible tuning options.
+
+
+=== Minimal requirements
+
+To configure {serverlessproductname} Serving for a minimal workload scenario, it is important to know the idle consumption of the system components.
+
+==== Idle consumption
+The idle consumption is dependent on the number of Knative Services.
+Please note, that for the idle consumption, only Memory is important.
+Relevant cycles of CPU are only used when Knative Services are changed or requests are sent to/from them.
+The following memory consumptions have been measured for the components in the `knative-serving` and `knative-serving-ingress` {product-title} projects:
+
+|===
+| Component | 0 Services | 100 Services | 500 Service | 1000 Services
+
+| activator
+| 55Mi
+| 86Mi
+| 150Mi
+| 200Mi
+
+| autoscaler
+| 52Mi
+| 102Mi
+| 225Mi
+| 350Mi
+
+| controller
+| 100Mi
+| 135Mi
+| 250Mi
+| 400Mi
+
+| webhook
+| 60Mi
+| 60Mi
+| 60Mi
+| 60Mi
+
+| 3scale-kourier-gateway (1)
+| 20Mi
+| 60Mi
+| 190Mi
+| 330Mi
+
+| net-kourier-controller (1)
+| 90Mi
+| 170Mi
+| 340Mi
+| 430Mi
+
+| istio-ingressgateway (1)
```

Comment:
There are more stuff to consider when it comes to traffic it seems (https://www.linkedin.com/feed/update/urn:li:activity:7114044569297555456/)  :thinking: 
I think for Istio we need some pointers about its performance.

Reply:
Agreed, probably need to talk to Lukas to get something from SM here. In the end its not up to us to document the specifics for SM.

Reply:
One thing what we might check in the future is if we can optimize envoy for Kourier, but that is another topic.